sound/soc/sof/intel/tgl.c

```diff
@@ -50,6 +50,24 @@ static int tgl_dsp_core_put(struct snd_sof_dev *sdev, int core)
 	return 0;
 }
 
+static int tgl_ipc4_dsp_core_put(struct snd_sof_dev *sdev, int core)
+{
+	const struct sof_ipc_pm_ops *pm_ops = sdev->ipc->ops->pm;
+	int ret;
+
+	if (pm_ops->set_core_state) {
+		ret = pm_ops->set_core_state(sdev, core, false);
+		if (ret < 0)
+			return ret;
+	}
+
+	/* power down primary core and return */
+	if (core == SOF_DSP_PRIMARY_CORE)
+		return hda_dsp_core_reset_power_down(sdev, BIT(core));
+
+	return 0;
```

Comment:
why can't we use the same sequence for IPC4 and IPC3? Would the IPC3 part be broken? If the firmware doesn't do anything in the set_core_state, that's fine, no?

Reply:
set_core_state do nothing for primary core in ipc3 path, so we can use the single sequence. I will change it. Thanks!